data-collection/deploy/module-rds-multitenant.yaml

```diff
@@ -0,0 +1,677 @@
+
+              print(f"Total metrics processed for instance {instance_id}: {len(all_flattened_metrics)}")
+              return all_flattened_metrics
+
+
+          def assume_role(account_id, service, region):
+              partition = boto3.session.Session().get_partition_for_region(region_name=region)
+              role_name = os.environ['ROLENAME']
+              assumed = boto3.client('sts', region_name=region).assume_role(
+                  RoleArn=f"arn:{partition}:iam::{account_id}:role/{role_name}",
+                  RoleSessionName='rds_multitenant_data_collection'
+              )
+              creds = assumed['Credentials']
+              return boto3.client(service, region_name=region,
+                  aws_access_key_id=creds['AccessKeyId'],
+                  aws_secret_access_key=creds['SecretAccessKey'],
+                  aws_session_token=creds['SessionToken']
+              )
+
+          def write_metrics_to_s3(s3_client, region_metrics, account_id, payer_id):
+
+              # Group metrics by their individual timestamps
+              for region, metrics in region_metrics.items():
+                  if not metrics:
+                      print(f"No metrics to process for region {region}")
+                      continue
+
+                  # Group metrics by their unique timestamp keys
+                  timestamp_grouped_metrics = {}
+                  for metric in metrics:
+                      timestamp = datetime.strptime(metric['timestamp'], "%Y-%m-%d %H:%M:%S%z")
+                      year = timestamp.strftime('%Y')
+                      month = timestamp.strftime('%m')
+                      day = timestamp.strftime('%d')
+                      hour = timestamp.strftime('%H')
+
+                      # Create a unique key for each timestamp
+                      timestamp_key = f"{year}/{month}/{day}/{hour}"
+                      
+                      if timestamp_key not in timestamp_grouped_metrics:
+                          timestamp_grouped_metrics[timestamp_key] = []
+                      
+                      timestamp_grouped_metrics[timestamp_key].append(metric)
+
+                  # Write metrics for each unique timestamp
+                  for timestamp_key, grouped_metrics in timestamp_grouped_metrics.items():
+                      year, month, day, hour = timestamp_key.split('/')
+
+                      # Create S3 key for this specific timestamp group with payer_id partitioning
+                      s3_key = f"{metrics_s3_prefix}/payer_id={payer_id}/account_id={account_id}/region={region}/year={year}/month={month}/day={day}/hour={hour}/metrics.parquet"
+
+                      print(f"Writing metrics to S3 key: {s3_key}")
+                      print(f"Total number of metrics: {len(grouped_metrics)}")
+
+                      # Convert to Arrow table
+                      table = pa.Table.from_pylist(grouped_metrics)
+
+                      # Write to Parquet format
```

Comment:
do we need to write to parquert format? Why not just json? Surely not as performant but no dependency needed